spec/javascripts/controllers/account_controller_spec.coffee

```diff
@@ -214,3 +222,23 @@ do ->
       it 'calls ShortFormApplicationService.importUserData', ->
         scope._createAccountSubmitApplication()
         expect(fakeShortFormApplicationService.submitApplication).toHaveBeenCalled()
+
+    describe '$scope.isSale', ->
+      it 'calls ListingIdentityService.isSale', ->
+        spyOn(fakeListingIdentityService, 'isSale')
+        scope.isSale(fakeListing)
+        expect(fakeListingIdentityService.isSale).toHaveBeenCalled()
+
+    describe '$scope.hasSaleAndRentalApplications', ->
+      it 'calls ListingIdentityService.isSale', ->
+        spyOn(fakeListingIdentityService, 'isSale')
+        scope.hasSaleAndRentalApplications([fakeApplication])
+        expect(fakeListingIdentityService.isSale).toHaveBeenCalled()
+      it 'returns false if there are no applications', ->
+        expect(scope.hasSaleAndRentalApplications([])).toEqual(false)
+      it 'returns true if there are at least two different applications', ->
+        fakeListingIdentityService.isSale = ->
+          fakeIsSale = !fakeIsSale
```

Comment:
so just to confirm this is saying first return true, then false? I'm not super clear on the mocking syntax and still find this a bit confusing

Reply:
I didnt have the better idea, how to mock it. Essentially, it toggles between true and false with every request. I can leave a comment there if you like.

Reply:
@software-project in a test i wrote that needed `isSale` to return true once and false once, i used multiple return values for a jasmine spy - see here: https://github.com/Exygy/sf-dahlia-web/blob/b1d8535b21579405ea306f793f862329825e9ddd/spec/javascripts/components/listing/listingContainer_spec.coffee#L129. would this be helpful here?

Reply:
that would be a bit clearer to me! thanks for the example @callaghanc !

Reply:
Thanks @callaghanc, that looks better.

Reply:
@software-project did you push this update?

Reply:
I did: https://github.com/Exygy/sf-dahlia-web/blob/aa4bf4d0367b51d135c14348220747748303159d/spec/javascripts/components/listing/listingContainer_spec.coffee#L129

Reply:
sorry, my bad. I thought @callaghanc updated it. Will do that now.